src/Examine.Lucene/Suggest/AnalyzingInfixSuggesterDefinition.cs

```diff
@@ -0,0 +1,93 @@
+using Lucene.Net.Analysis;
+using Lucene.Net.Index;
+using Lucene.Net.Search.Spell;
+using System.Linq;
+using Lucene.Net.Search.Suggest;
+using Lucene.Net.Search.Suggest.Analyzing;
+using Lucene.Net.Util;
+using Examine.Suggest;
+using static Lucene.Net.Search.Suggest.Lookup;
+using System.Collections.Generic;
+
+namespace Examine.Lucene.Suggest
+{
+    public class AnalyzingInfixSuggesterDefinition : LuceneSuggesterDefinition, ILookupExecutor
+    {
+        public AnalyzingInfixSuggesterDefinition(string name, string[] sourceFields = null, ISuggesterDirectoryFactory directoryFactory = null, Analyzer queryTimeAnalyzer = null)
+            : base(name, sourceFields, directoryFactory,queryTimeAnalyzer)
+        {
+        }
+        /// <inheritdoc/>
+        public Lookup Lookup { get; internal set; }
+
+        /// <inheritdoc/>
+        public override ILookupExecutor BuildSuggester(FieldValueTypeCollection fieldValueTypeCollection, ReaderManager readerManager, bool rebuild)
+            => BuildAnalyzingInfixSuggesterLookup(fieldValueTypeCollection, readerManager, rebuild);
+
+        /// <inheritdoc/>
+        public override ISuggestionResults ExecuteSuggester(string searchText, ISuggestionExecutionContext suggestionExecutionContext) => ExecuteAnalyzingInfixSuggester(searchText, suggestionExecutionContext);
+
+        protected ILookupExecutor BuildAnalyzingInfixSuggesterLookup(FieldValueTypeCollection fieldValueTypeCollection, ReaderManager readerManager, bool rebuild)
+        {
+            string field = SourceFields.First();
```

Comment:
What happens if `SourceFields` are null or empty? I can see that the constructor on line 16 passes null as `sourceFields`

Reply:
Added checks to constructor and inline. Changed constructor signature to remove null defaults.